test.js

```diff
@@ -241,5 +246,37 @@ describe('shopify-token', function () {
         done();
       });
     });
+
+    it('returns an error if timeout is exceeded (no response)', function (done) {
+      scope
+        .post(pathname)
+        // Delay response header. (Client has 10ms timeout above.)
+        .delay({ head: 100 })
```

Comment:
Interestingly, changing this to `delaySocket` makes it pass. (See https://github.com/node-nock/nock#delay-the-connection.) So I think my implementation of `request.setTimeout` isn't quite right. But as noted in https://github.com/lpinca/shopify-token/issues/4, I think reinventing the wheel on this isn't worth the time; it would be better to use a request abstraction module that has already figured it out.